tests/test_app/test_data_store_new.py

```diff
@@ -143,6 +144,38 @@ def full_dstore(write_dir, nc_objects, completed_objects, log_data):
     dstore.write_log(unique_id="scitrack.log", data=log_data)
     return dstore
 
+@pytest.fixture(scope="function")
+def tinydbfile_locked(tmp_dir):
+    path = tmp_dir / "sample_locked.tinydb"
+    shutil.copy(DATA_DIR / path.name, path, )
+    return path
+
+
+@pytest.fixture(scope="function")
+def tinydbfile_notlocked(tmp_dir):
+    try:
+        from fnmatch import fnmatch, translate
+        from tinydb import Query, TinyDB
+        from tinydb.middlewares import CachingMiddleware
+        from tinydb.storages import JSONStorage
+        from cogent3.app.data_store import load_record_from_json
+        from cogent3.app.sqlite_data_store import DataStoreSqlite
+    except ImportError as e:
+        raise ImportError(
+            "You need to install tinydb to be able to migrate to new datastore."
+        ) from e
+    path = tmp_dir / "sample_locked.tinydb"
```

Comment:
1 - The above rename you mentioned, renames on its own name. 
2 - When I use `tinydbfile_locked` fixture here instead of copying it, I don't have this line "path = tmp_dir / "sample_locked.tinydb"
3 - Now, I will use `rename` in defining the other fixture `tinydbfile_locked`, instead of `shutil.copy`.

Reply:
There's a typo in my statement, it should say "sample_**un**locked.tinydb".

You need the `shutil.copy()`! Here's the process

1. copy the versioned "sample_locked.tinydb" file into a temporary directory
2. rename the copy to unlocked so at least the name reflects its state (that's the rename statement with "sample_unlocked.tinydb")
3. remove the lock
4. return the path to the renamed file